### PR TITLE
Update NextAuth mocking method

### PR DESCRIPTION
**Pull request by alan910127**
Change to a method that can directly change auth state without modifying environment variables

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,13 +10,12 @@ Score management made easy
...
 
 Commit message: prefix with an [meaningful emoji](https://gitmoji.dev)
 
-## Mocking Next Auth during Development
+## Mocking NextAuth during Development
 
-To mock NextAuth in development, you should set the environment variable `NEXT_PUBLIC_MOCK_ROLE` to `admin`, `ta` or `student` in your `.env` file.
-
-⚠️ Switching between mocking or not needs manual refresh on the client side. ⚠️
+To mock NextAuth in development, you should set the environment variable `NEXT_PUBLIC_MOCK_NEXTAUTH` to `true` in your `.env` file.
```
Comment (alan910127) [inline README.md:21]: Fixed mock NextAuth, so it can just specify enable or not now

File: src/mock/auth/callbacks.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      if (!isCredentialCallback(req, providerId)) {
+        return encode(params);
+      }
+      const cookies = new Cookies(req, res);
+      const sessionToken = cookies.get(COOKIES_SESSION_TOKEN);
+
+      if (sessionToken == null) return "";
+
+      const user = (await adapter.getSessionAndUser(sessionToken))?.user;
+      if (!user || params.token?.sub == null || user.id === params.token.sub) {
```
Comment (alan910127) [inline src/mock/auth/callbacks.ts:95]: `params.token.sub` is the id of the requested user, maybe we could extract it to a variable for readability.

File: src/mock/mockUsers.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    id: "1",
+    name: "TA",
+    email: "ta@example.com",
+  },
+  student: {
+    id: "2",
+    name: "Student",
+    email: "student@example.com",
+  },
+};
```
Comment (alan910127) [inline src/mock/mockUsers.ts:23]: This file is just renamed from `mockSession.ts`

File: src/mock/mockUsers.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import type { User } from "next-auth";
```
Comment (alan910127) [inline src/mock/mockUsers.ts:1]: This file is renamed from `mockSession.ts`

**Review by alan910127 — COMMENTED** [review]
Improve the NextAuth mocking, and do a little refactor

**Review by BaBa0525 — APPROVED** [review]
超級狠

**Pull request merged**